### PR TITLE
GRIDEDIT-785 Refactoring of the averaging strategy

### PR DESCRIPTION
**Pull request by BillSenior**
This branch contains only the refactoring of the averaging strategies

### CODE REVIEW

File: libs/MeshKernel/include/MeshKernel/AveragingStrategies/AveragingStrategy.hpp [hunk 1]
```diff
@@ -36,13 +36,13 @@ namespace meshkernel::averaging
...
-        virtual void Add(Point const& samplePoint, double sampleValue) = 0;
-
-        /// @brief Calculates the average value based on the values added.
+        /// @brief Calculates the average value based on the sample values.
+        /// @param[in] interpolationPoint The point for which the average should be calculated.
+        /// @param[in] samplePoints The sample points to used by this strategy.
+        /// @param[in] sampleValues The sample values  associated with each sample point.
         /// @return The calculated average
-        [[nodiscard]] virtual double Calculate() const = 0;
+        virtual double Calculate(const Point& interpolationPoint,
```
[the remark this answers is not in the document]
Comment (BillSenior) [reply libs/MeshKernel/include/MeshKernel/AveragingStrategies/AveragingStrategy.hpp:44]: done

File: libs/MeshKernel/src/AveragingInterpolation.cpp [hunk 2]
```diff
@@ -92,7 +86,6 @@ void AveragingInterpolation::Compute()
     if (m_interpolationLocation == Location::Edges)
     {
         m_edgeResults.resize(m_mesh.GetNumEdges(), constants::missing::doubleValue);
-        std::ranges::fill(m_edgeResults, constants::missing::doubleValue);
```
[the remark this answers is not in the document]
Comment (BillSenior) [reply libs/MeshKernel/src/AveragingInterpolation.cpp]: done

File: libs/MeshKernel/src/AveragingInterpolation.cpp [hunk 3]
```diff
@@ -110,11 +103,10 @@ void AveragingInterpolation::Compute()
 
     if (m_interpolationLocation == Location::Faces)
     {
+        std::vector<bool> visitedSamples(m_samples.size(), false); ///< The visited samples
+        std::vector<Point> polygonNodesCache(Mesh::m_maximumNumberOfNodesPerFace + 1);
         m_faceResults.resize(m_mesh.GetNumFaces(), constants::missing::doubleValue);
-        std::ranges::fill(m_faceResults, constants::missing::doubleValue);
```
[the remark this answers is not in the document]
Comment (BillSenior) [reply libs/MeshKernel/src/AveragingInterpolation.cpp:114]: done

File: libs/MeshKernel/include/MeshKernel/AveragingStrategies/AveragingStrategy.hpp [hunk 4]
```diff
@@ -36,13 +36,13 @@ namespace meshkernel::averaging
...
+                                 const std::vector<Point>& samplePoints,
```
[the remark this answers is not in the document]
Comment (BillSenior) [reply libs/MeshKernel/include/MeshKernel/AveragingStrategies/AveragingStrategy.hpp]: Changed to array of Samples

**Review by ahmad-el-sayed — APPROVED** [review]

**Pull request merged**